ProcessOptimizer/tests/test_space.py

```diff
@@ -677,31 +707,42 @@ def test_purely_categorical_space():
 # =============================================================================
 
 
-@pytest.mark.fast_test
-def test_lhs_arange():
-    # Test that the latin hypercube sampling functions run normally
-    dim_cat = Categorical(["a", "b", "c"])
-    dim_cat.lhs_arange(10)
-    dim_int = Integer(-10, 20)
-    lhs_int = dim_int.lhs_arange(10)
-    dim_real = Real(-10, 20)
-    lhs_real = dim_real.lhs_arange(10)
-    # Test that the Integer and Real classes return identical points after
-    # rounding, using Numpy's allclose function
-    assert np.allclose(lhs_int, lhs_real, atol=0.5)
-    # Test that the Integer class returns values that are ints for all intents 
-    # and purposes
-    assert all([np.mod(x,1) == 0 for x in lhs_int])
-    # Test that the Real class returns flots in all locations
-    assert all([isinstance(x,np.float64) for x in lhs_real])
-
-
 @pytest.mark.fast_test
 def test_lhs():
     SPACE = Space(
-        [Integer(-20, 20), Real(-10.5, 100), Categorical(list("abc"))]
+        [
+            Integer(1, 6),
+            Real(1, 7),
+            Real(10**-3, 10**3, prior="log-uniform"),
+            Categorical(list("abc")),
+        ]
     )
-    samples = SPACE.lhs(10)
-    assert len(samples) == 10
-    assert len(samples[0]) == 3
-
+    samples = SPACE.lhs(6)
+    assert len(samples) == 6
+    assert len(samples[0]) == 4
+    values = [[sample[i] for sample in samples] for i in range(4)]
+    # set disregards order, so we use it to test which values we got.
+    assert set(values[0]) == set(range(1, 6 + 1))
+    # We should get all six integers, in any order
+    assert set(values[1]) == set(0.5 + np.arange(1, 7))
+    # Six reals between 1 and 7, evenly divide, lie halfway between integers
+    for log_real in values[2]:
+        assert any(
+            log_real == pytest.approx(value)
+            for value in np.sqrt(10) * 10**-3 * 10 ** np.arange(6)
+        )
+    # there are no good tests for approximate equality that disregards order
+    # For log-uniform, we should get six values evenly spaced on a log scale
+    assert set(values[3]) == set("abc")
+    is_min_pos = [[y == min(x) for y in x] for x in values]
+    num_min_pos = np.where(is_min_pos[0:3])[1]
+    # We only bother for the numerical axes, since the categorical axis has fewer than six values, so it has repeated values.
+    assert len(np.unique(num_min_pos)) > 1
+    # The minimum value should not always be in the same position
+    lhs_one = SPACE.lhs(n=6, random_state=None)
+    lhs_two = SPACE.lhs(n=6, random_state=None)
+    assert lhs_one != lhs_two
+    # Verfying that lhs can be performed randomly
+    for i in range(4):
+        assert set([x[i] for x in lhs_one]) == set([x[i] for x in lhs_two])
+    # Asserting the the values are the same for both the lhs, even though the order is different
```

Comment:
```suggestion
    # Asserting the the values are the same for both the lhs, even though the order is different
    for i in range(4):
        assert set([x[i] for x in lhs_one]) == set([x[i] for x in lhs_two])    
```
Moving comment up above the code it refers to for consistency.